src/main/java/games/strategy/engine/pbem/TripleAForumPoster.java

```diff
@@ -41,33 +41,38 @@ public boolean postTurnSummary(final String summary, final String title) {
       final String token = getToken(client, userId);
       try {
         post(client, token, "### " + title + "\n" + summary);
-        m_turnSummaryRef = "Sucessfully posted!";
+        turnSummaryRef = "Sucessfully posted!";
         return true;
       } finally {
         deleteToken(client, userId, token);
       }
-    } catch (final Exception e) {
+    } catch (final IOException | IllegalStateException e) {
       ClientLogger.logQuietly("Failed to post game to forum", e);
-      m_turnSummaryRef = e.getMessage();
+      turnSummaryRef = e.getMessage();
     }
     return false;
   }
 
-  private void post(final CloseableHttpClient client, final String token, final String text) throws Exception {
+  private void post(final CloseableHttpClient client, final String token, final String text) throws IOException {
     final HttpPost post = new HttpPost(tripleAForumURL + "/api/v1/topics/" + getTopicId());
     addTokenHeader(post, token);
     post.setEntity(new UrlEncodedFormEntity(
         Collections.singletonList(new BasicNameValuePair("content",
-            text + ((m_includeSaveGame && m_saveGameFile != null) ? uploadSaveGame(client, token) : ""))),
+            text + ((m_includeSaveGame && saveGameFile != null) ? uploadSaveGame(client, token) : ""))),
         StandardCharsets.UTF_8));
     HttpProxy.addProxy(post);
-    client.execute(post).close();
+    try (CloseableHttpResponse response = client.execute(post)) {
+      final int code = response.getStatusLine().getStatusCode();
+      if (code != HttpURLConnection.HTTP_OK) {
+        throw new IllegalStateException("Forum responded with code " + code);
```

Comment:
The response body could have some very useful details, we should probably log that in addition to status code. For most users a status code of 500 or 503 or what not will not be very meaningful (and wont' be very meaningful for dev's either)